DEV-CONVENTIONS.md

```diff
@@ -36,6 +36,8 @@ Others docs you should look at:
 
       - Must use lowerCamel case for Go package names.
```

Comment:
Can we also enforce that singular nouns should be used of plural? We've been de-facto following that convention, but it doesn't seem to be mentioned here.

Reply:
@kyessenov I think it is hard to enforce since sometimes a type represents a group of objects, like in the following case:

https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/nestedpendingoperations/nestedpendingoperations.go

```go
type nestedPendingOperations struct {
	operations                []operation
	...
}```

Reply:
Interesting. Maybe we need to phrase it carefully, or simply advise it instead. For example, mixer adapters are in package `adapter`, because each adapter is an element in a collection, rather than a collection itself.

Reply:
@kyessenov This reminded me reducing points in a programming homework in university, being a TA ten years ago: https://webcourse.cs.technion.ac.il/234122/Winter2007-2008/hw_Homework%20Updates_ex3.html :)

> Confusing singular/plural of the nouns : calling a single movie Movies, Movies_list etc. or calling a collection of movies Movie.